doc/unison-manual.tex

```diff
@@ -1995,7 +1995,9 @@
 monitoring), interrupting with ``Ctrl-C'' or with signal \verb|SIGINT| or
 \verb|SIGTERM| works the same way as described above and will additionally stop
 the continuous process. To stop only the continuous process and let the last
-synchronization complete normally, send signal \verb|SIGUSR2| instead.
+synchronization complete normally, send signal \verb|SIGUSR2| or \verb|SIGQUIT|
+(the latter can also be generated from keyboard by pressing
+``Ctrl-\textbackslash'' or ``Ctrl-4''), instead.
```

Comment:
I'm not entirely comfortable with this because SIGQUIT means "exit and dump core".   I know nobody does that very much any more.

I do wonder about a single SIGINT starting graceful shutdown, and a second one doing what SIGINT does now.  That seems to be a common idiom, and therefore aligned with the Principle of Least Astonishment.

Reply:
I was hesitant about SIGQUIT. The only reason I added it was because it can be generated from a keypress (but that brings its own issues).

Can't really use SIGINT the way you propose because... Windows. Windows does not really support signals, but it has a few and SIGINT is one of them. Using SIGINT for graceful shutdown would not work because issue #812 can't be solved in Windows AFAICT. #812 is solved for POSIX platforms and if someone figures out how to solve it for Windows then SIGINT could be considered for graceful shutdown.

As for the common idiom, to me the common idiom is what was implemented in #810. The first SIGINT will cause a termination as quickly as possible but it will do an orderly shutdown, trying to do cleanup with the server (all this can still take considerable time). Repeating SIGINT will then do an immediate exit, ignoring any cleanup.

Using the first SIGINT for the graceful shutdown as presented by this PR would not be the common idiom in my mind because the stop can be requested at any time during an ongoing sync and it will first react after the sync. So the sync may very well continue for another hour or more. Only once the sync is complete will the process actually stop. It's not really an interruption and it can take a lot of time before finally terminating. It's a bit difficult to associate with SIGINT.

Reply:
One more thing about SIGINT. This PR is about stopping the repeat mode only. SIGINT works even without repeat mode. It's better that SIGINT always behaves the same, repeat mode or not.